status-im/status-react
# Add Wallet Connect base implementation of the connection flow

fixes #19907

### Summary

In order to not diverge/conflict when implementing the features, we should have a starting point from which we will add the functionality on top from other user stories. Currently connection only works with Optimism mainnet.

### Steps to test (developers)


1. Make sure the wallet connect is enabled via feature flags (set `FLAG_WALLET_CONNECT_ENABLED` to `1` in the `.env.debug` file or hardcode the default value)
2. Add `WALLET_CONNECT_PROJECT_ID` to .env.debug (ask for the ID in the #wallet-connect Discord channel) or hardcode the value
3. Go to https://react-app.walletconnect.com/
4. Un-toggle "Testnets Only?"
5. Select only Optimism and press connect at the bottom
6. Copy the uri and run in the REPL: `(rf/dispatch [:wallet-connect/pair uri])`
7. When successful, run: `(rf/dispatch [:wallet-connect/approve-session])`
8. The connection should be established


status: ready
